plugin.video.retrospect/retroservice.py

```diff
@@ -0,0 +1,144 @@
+
+    # If there was an old profile, migrate it.
+    old_profile = os.path.abspath(os.path.join(new_profile, "..", old_add_on_id))
+    if not os.path.exists(old_profile):
+        xbmc.log("{}: No old profile data found at {}.".format(add_on_name, old_profile), log_level)
+        return
+    xbmc.log("{}: Old Profile located at {}".format(add_on_name, old_profile), log_level)
+
+    d.notification(add_on_name, "Migrating add-on data to new format.", icon="info", time=5)
+
+    xbmc.log("{}: Migrating addon_data {} to {}.".format(
+        add_on_name, old_profile, new_profile), log_level)
+    shutil.copytree(old_profile, new_profile, ignore=shutil.ignore_patterns("textures"))
+
+    # If there were local setttings, we need to migrate those too so the channel ID's are updated.
+    local_settings_file = os.path.join(new_profile, "settings.json")
+    if os.path.exists(local_settings_file):
+        xbmc.log("{}: Migrating {}.".format(add_on_name, local_settings_file), log_level)
+        with io.open(local_settings_file, mode="rb") as fp:
+            content = fp.read()
+            settings = json.loads(content, encoding='utf-8')
+
+        channel_ids = settings.get("channels", {})
+        channel_settings = {}
+        for channel_id in channel_ids:
+            new_channel_id = channel_id.replace(old_add_on_id, add_on_id)
+            xbmc.log("{}: - Renaming {} -> {}.".format(
+                add_on_name, channel_id, new_channel_id), log_level)
+            channel_settings[new_channel_id] = settings["channels"][channel_id]
+
+        settings["channels"] = channel_settings
+        with io.open(local_settings_file, mode='w+b') as fp:
+            content = json.dumps(settings, indent=4, encoding='utf-8')
+            fp.write(content)
+
+    # fix the favourites
+    favourites_path = os.path.join(new_profile, "favourites")
+    if os.path.isdir(favourites_path):
+        xbmc.log("{}: Updating favourites at {}".format(add_on_name, favourites_path), log_level)
+        for fav in os.listdir(favourites_path):
+            # plugin://net.rieter.xot/
+            fav_path = os.path.join(favourites_path, fav)
+            xbmc.log("{}: - Updating favourite: {}.".format(add_on_name, fav), log_level)
+            with io.open(fav_path, mode='r', encoding='utf-8') as fp:
+                content = fp.read()
+
+            content = content.replace("plugin://{}/".format(old_add_on_id),
+                                      "plugin://{}/".format(add_on_id))
+            with io.open(fav_path, mode='w+', encoding='utf-8') as fp:
+                fp.write(content)
+
+    xbmc.log("{}: Migration completed.".format(add_on_name), log_level)
+    return
+
+
+def autorun_retrospect():
+    if xbmcaddon.Addon().getSetting("auto_run") == "true":
+        xbmc.executebuiltin('RunAddon(plugin.video.retrospect)')
+    return
+
+
+# Check for the migration
+migrate_profile(Config.profileDir, Config.addonId, Config.rootDir, Config.appName)
```

Comment:
what is this?

Reply:
I renamed my add-on a couple of months ago. This renames the corresponding addon_data folder, for those users that are moving from old to new naming. That way they keep their settings.

Reply:
Maybe it can be already dropped since the big majority of your users are likely on your last version

Reply:
Dropped it.